.github/workflows/build_and_upload_artifact.yml

```diff
@@ -0,0 +1,58 @@
+name: Build and Upload Artifact
+
+on:
+    push:
+        branches: [ main ]
+    pull_request:
+        branches: [ main ]
+
+jobs:
+    build:
+        runs-on: ubuntu-latest
+
+        container:
+            image: debian:bookworm
+
+        steps:          
+        - name: Install dependencies
+          run: |
+                apt-get update
+                apt-get install -y git cmake glslang-dev glslang-tools graphviz libgbm-dev libinput-dev libudev-dev libvulkan-dev \
+                libxcb-composite0-dev libxcb-dri3-dev libxcb-ewmh-dev libxcb-icccm4-dev libxcb-present-dev libxcb-render-util0-dev \
+                libxcb-res0-dev libxcb-xinput-dev libxkbcommon-dev meson plantuml xmlto xsltproc wayland-protocols libwlroots-dev doxygen \
+                libexpat1-dev libexpat1 libxml2-dev hwdata libcairo2-dev libncurses-dev
+
+        - name: Checkout code
+          uses: actions/checkout@v2
+        
+        - name: Checkout submodules
+          run: |
+               git config --global --add safe.directory /__w/wlmaker/wlmaker
+               git submodule update --init --recursive --merge
+
+        - name: Build dependencies
+          run: |
+               cd dependencies
+               export LD_LIBRARY_PATH="${HOME}/.local/lib/x86_64-linux-gnu"
```

Comment:
optional: portability issue -- you might change this to ${MACHTYPE}  (in bash).

(not sure it's worth it; but might be helpful to add a comment about it, so if somebody ever runs it on another architecture, they have a head-start in fixing)

Reply:
Thanks, on it.

Reply:
`${MACHTYPE}` seems to resolve to `x86_64-pc-linux-gnu`, whereas we need `x86_64-linux-gnu`.

`dpkg-architecture -qDEB_HOST_MULTIARCH` seems to give us what we need, though.

Reply:
Oh, that's a subtle detail..! Well spotted & thanks for fixing here!